### ISSUE TITLE
Request rate is large.

### ISSUE THREAD

**Issue by vipinsorot** [AzureCosmosDB/data-migration-desktop-tool#83]
**Error-msg**: Data transfer failed
      Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: TooManyRequests (429); Substatus: 3200; ActivityId: cfe3d2b9-8315-4ce4-9833-2ef94a6f7d82; Reason: (
      code : TooManyRequests
      message : Message: {"Errors":["Request rate is large. More Request Units may be needed, so no changes were made. Please retry this request later. Learn more: http://aka.ms/cosmosdb-error-429"]}

release:2.1.3

**Comment by joelhulen** [comment]
You can handle this in a couple of ways:

1. Temporarily increase your Cosmos DB container (or database) scale settings by increasing the number of Request Units per second (RU/s) for the duration of your data transfer.
2. Go to your Cosmos DB sink settings and increase the `MaxRetries` value (default 5). This will tell the Polly retry policy to wait and retry `n` times, based on the setting.

This is where the retry policy is defined: https://github.com/AzureCosmosDB/data-migration-desktop-tool/blob/f45805454bf824b163ee166f4982ac2994560447/Extensions/Cosmos/Cosmos.DataTransfer.CosmosExtension/CosmosDataSinkExtension.cs#L98

**Comment by vipinsorot** [comment]
@joelhulen 
yup i do have converted my collection configuration from manual to auto scale for RU/s
can you plz share an example related to retry mechanism ?

**Comment by vipinsorot** [comment]
my migration file
{
  "Source": "cosmos-nosql",
  "Sink": "cosmos-nosql",
  "SourceSettings": {
    "ConnectionString": "AccountEndpoint=****",
    "Database": "INTEGRATION",
    "Container": "Invoice",
    "PartitionKeyValue": "/data/dealClaimId",
    "Query": "SELECT * FROM c"
  },
  "SinkSettings": {
    "ConnectionString": "AccountEndpoint=***",
    "Database": "INTEGRATION",
    "Container": "Invoice",
    "BatchSize": 100,
    "MaxRetryCount": 5,
    "RecreateContainer": false,
    "ConnectionMode": "Gateway",
    "CreatedContainerMaxThroughput": 1000,
    "UseAutoscaleForCreatedContainer": true,
    "InitialRetryDurationMs": 200,
    "WriteMode": "InsertStream",
    "IsServerlessAccount": false,
    "PartitionKeyPath": "/data/dealClaimId"
  },
  "Operations": []
}

**Comment by joelhulen** [comment]
I just saw your last message. Increase `MaxRetryCount` under **SinkSettings** to something like 20, in addition to my notes on a potential auto-scale increase.

**Comment by vipinsorot** [comment]
Yup i have already increased the RU/s to 10k for both source and sink

**Comment by vipinsorot** [comment]
https://github.com/AzureCosmosDB/data-migration-desktop-tool/blob/f45805454bf824b163ee166f4982ac2994560447/Extensions/Cosmos/Cosmos.DataTransfer.CosmosExtension/CosmosDataSourceExtension.cs#L36
call is updating **feedIterator.HasMoreResults** to **false**, has led to fetching a total of 0 records.

**Comment by vipinsorot** [comment]
@joelhulen  it worked post refactoring the code
![image](https://github.com/AzureCosmosDB/data-migration-desktop-tool/assets/130037251/a61a814a-ef24-45f5-9111-4263964fc2ae)
